drv/i2c-devices/src/pca9956b.rs

```diff
@@ -0,0 +1,332 @@
+        let buffer = [reg as u8, val];
+        self.device
+            .write(&buffer)
+            .map_err(|code| Error::I2cError(code))
+    }
+
+    /// Write a number of Registers into `buf`
+    /// Note that CTRL_AUTO_INCR_MASK is set here to enable the device's
+    /// auto-increment feature, which has varying behaviors and limitations that
+    /// must be accounted for.
+    fn write_buffer(&self, reg: Register, buf: &[u8]) -> Result<(), Error> {
+        let mut data: [u8; 25] = [0; 25];
+        data[0] = (reg as u8) | CTRL_AUTO_INCR_MASK;
+        data[1..=buf.len()].copy_from_slice(buf);
+
+        self.device
+            .write(&data[..=buf.len()])
+            .map_err(|code| Error::I2cError(code))
+    }
+
+    /// Sets the device's IREFALL register to `val`
+    /// IREFALL gets copied into all IREFn registers by the device.
+    pub fn set_iref_all(&self, val: u8) -> Result<(), Error> {
+        self.write_reg(Register::IREFALL, val)
+    }
+
+    /// Sets the device's PWMALL register to `val`
+    /// PWMALL gets copied into all PWMn registers by the device.
+    pub fn set_pwm_all(&self, val: u8) -> Result<(), Error> {
+        self.write_reg(Register::PWMALL, val)
+    }
+
+    /// Sets the PWMx register to `val` for a given `led`
+    pub fn set_a_led_pwm(&self, led: u8, val: u8) -> Result<(), Error> {
+        if led >= NUM_LEDS as u8 {
+            return Err(Error::InvalidLED(led));
+        }
+        let reg = FromPrimitive::from_u8((Register::PWM0 as u8) + led).unwrap();
+        self.write_reg(reg, val)
+    }
+
+    /// Sets the PWMx register a number of LEDs, up to NUM_LEDS, beginning with
+    /// PWM0.
+    pub fn set_all_led_pwm(&self, vals: &[u8]) -> Result<(), Error> {
+        if vals.len() > NUM_LEDS {
+            return Err(Error::InvalidLED(
+                vals.len().try_into().unwrap_or(0xFF),
+            ));
+        }
+        self.write_buffer(Register::PWM0, &vals)
+    }
+
+    /// Queries the MODE2 register and to check the OVERTEMP and ERROR bits
+    /// If ERROR is set, each EFLAGx register will be read and parsed.
+    /// An important thing to note about this device is that in order to do
+    /// error detection (reflected in EFLAGx) the PWMx associated with the LED
+    /// must be higher than 0x08. So if the PWMx is set to zero, not error can
+    /// be detected.
+    pub fn check_errors(&self) -> Result<LedErrSummary, Error> {
+        let mode2 = self.read_reg(Register::MODE2)?;
+        let overtemp = (mode2 & MODE2_OVERTEMP_MASK) != 0;
+        let error = (mode2 & MODE2_ERROR_MASK) != 0;
+
+        let mut err_state = Pca9956BErrorState {
+            overtemp: overtemp,
+            ..Default::default()
+        };
+
+        if error {
+            let mut eflags: [u8; 6] = [0; 6];
+            for i in 0..eflags.len() {
+                // Notably, the auto-increment function does not apply to these
+                // registers, so they must be fetched individually
```

Comment:
😱

Reply:
Yeah...Table 6 in the datasheet outlines the different configurations for auto-incrementing, none of which operate above `0x3E`. It makes a bit of sense, since after that is just `PWMALL`, `IREFALL`, and `EFLAGn` anyway.